### ISSUE TITLE
Ensure entire expression is within math before expansion

### ISSUE THREAD

**Issue by mohkale** [tecosaur/LaTeX-auto-activating-snippets#16]
I've run into this strange issue where if I write a formula in org with laas-mode enabled, and then end a sentance after the math fragment the snippet is still activated.

For example in an org-mode before enter: `hello \( some-math \). `. The entire latex math fragment is wrapped in resulting in `\\dot{( nm = -3 \)}`.

**Comment by ymarco** [comment]
Can reproduce.
laas-mathp is misbehaving and returning t when the cursor is right after \).

@tecosaur what do you think? Does auctex's texmathp work in org mode? It seems to behave correctly is this situation.

**Comment by ymarco** [comment]
Workaround for ". ":

(aas-set-snippets 'laas-mode
  ". " nil)

You won't be able to use the \dot snippets, but at least you could end sentences.

**Comment by ymarco** [comment]
Waiting for @tecosaur (or anyone else who's experianced with org-mode math)

**Comment by tecosaur** [comment]
So the way that Org element works is that each "element" may extend a whitespace char beyond whe we'd consider them ending. E.g.

    Hello *there* you.
          ^------^

So we'd probably want to tweak `lass-org-mathp` to see if we're at the end of the element, and if so assume that we're actually just past it.

**Comment by lucasvreis** [comment]
> Can reproduce. laas-mathp is misbehaving and returning t when the cursor is right after \). @tecosaur what do you think? Does auctex's texmathp work in org mode? It seems to behave correctly is this situation.

In my own config I use texmathp with my org-mode snippets and it has been working very well, you don't need to load all of auctex since it is defined in a separate file.

**Comment by ymarco** [comment]
Yeah. I think the switch from texmathp to laas-mathp occured because texmathp misbehaved in org-mode $$, but nowadays the consensus is to only use \(\) so we can bring it back.
